### DISCUSSION TITLE
Keep SpineOpt Generic : `vom_cost` / `fuel_cost` vs. e.g. `connection_flow_cost`

### DISCUSSION THREAD

**Discussion by DillonJ** [spine-tools/SpineOpt.jl#863] [Ideas]
Hi folks,

I don't believe there is any difference between `vom_cos` and `fuel_cost` in terms of their function? They are both simply a coefficient on `unit_flow` in the objective function. Contrast this with `connection_flow_cost` which is more generic.

I think we should try and be more generic and consistent, and in this example., have instead, `unit_flow_cost`. 

Thoughts?

**Comment by jkiviluo** [comment]
Absolutely.

**Comment by mihlema** [comment]
I think fixed & variable operational costs or fuel costs do speak more to the standard modeler. But instead of repeating code, we could deal with this e.g. in pre-processing?

**Comment by DillonJ** [comment]
But it's completely at odds with the generic structure of SpineOpt which appeals to very many. I think there are two problems: 

1. They both do exactly the same thing
2. There are specific and not generic. The fuel cost in particular, I find myself using it a lot - to impose a cost on a flow and only 1 out of 10 times is the flow actually a fuel flow. It's unnecessarily specific. The vom_cost too. A single parameter that is more generic can handle both of these cases and many many more.

**Comment by mihlema** [comment]
Maybe I am nostalgic about them. I agree that they do exactly the same and code duplication is not necessarily great. I'm okay with the change, but it has to come with documentation. I think both, the parameter.md on `unit_flow_cost` and its template entry should mention these two possible use cases. Then it's easy enough for a new user to search for these terms and realize which parameter to use.

**Comment by mihlema** [comment]
Just coming back to this conversation:
As I think @DillonJ pointed out, it can be useful to be able to differentiate between those costs. Alternatively, this can of course also be done as well in pre-/post-processing. tbd

**Comment by clizbe** [comment]
Closing due to stale status.
